pints/_mcmc/_monomial_gamma_hmc.py

```diff
@@ -0,0 +1,507 @@
+            self._current_momentum = self._sample_momentum()
+
+            # First leapfrog position is the current sample in the chain
+            self._position = np.array(self._current, copy=True)
+            self._gradient = np.array(self._current_gradient, copy=True)
+            self._momentum = np.array(self._current_momentum, copy=True)
+
+            # Perform a half-step before starting iteration 0 below
+            self._momentum -= self._scaled_epsilon * self._gradient * 0.5
+
+        # Perform a leapfrog step for the position
+        self._position += self._scaled_epsilon * (
+            self._K_deriv(self._momentum, self._a, self._c, self._m)
+        )
+
+        # Ask for the pdf and gradient of the current leapfrog position
+        # Using this, the leapfrog step for the momentum is performed in tell()
+        self._ready_for_tell = True
+        return np.array(self._position, copy=True)
+
+    def current_log_pdf(self):
+        """ See :meth:`SingleChainMCMC.current_log_pdf()`. """
+        return -self._current_energy
+
+    def leapfrog_steps(self):
+        """
+        Returns the number of leapfrog steps to carry out for each iteration.
+        """
+        return self._n_frog_iterations
+
+    def leapfrog_step_size(self):
+        """
+        Returns the step size for the leapfrog algorithm.
+        """
+        return self._step_size
+
+    def _log_init(self, logger):
+        """ See :meth:`Loggable._log_init()`. """
+        logger.add_float('Accept.')
+
+    def _log_write(self, logger):
+        """ See :meth:`Loggable._log_write()`. """
+        logger.log(self._mcmc_acceptance)
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Monomial-Gamma Hamiltonian Monte Carlo'
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def set_leapfrog_steps(self, steps):
+        """
+        Sets the number of leapfrog steps to carry out for each iteration.
+        """
+        steps = int(steps)
+        if steps < 1:
+            raise ValueError('Number of steps must exceed 0.')
+        self._n_frog_iterations = steps
+
+    def set_leapfrog_step_size(self, step_size):
+        """
+        Sets the step size for the leapfrog algorithm.
+        """
+        a = np.atleast_1d(step_size)
+        if len(a[a < 0]) > 0:
+            raise ValueError(
+                'Step size for leapfrog algorithm must' +
+                'be greater than zero.'
+            )
+        if len(a) == 1:
+            step_size = np.repeat(step_size, self._n_parameters)
+        elif not len(step_size) == self._n_parameters:
+            raise ValueError(
+                'Step size should either be of length 1 or equal to the' +
+                'number of parameters'
+            )
+        self._step_size = step_size
+        self._set_scaled_epsilon()
+
+    def divergent_iterations(self):
```

Comment:
Could you move this up so the methods are ordered alphabetically?

Reply:
Functions now alphabetical (I think).